chia/util/keychain.py

```diff
@@ -132,6 +143,104 @@ def get_private_key_user(user: str, index: int) -> str:
     return f"wallet-{user}-{index}"
 
 
+@final
+@streamable
+@dataclass(frozen=True)
+class KeyDataSecrets(Streamable):
+    mnemonic: List[str]
+    entropy: bytes
+    private_key: PrivateKey
+
+    def __post_init__(self) -> None:
+        # This is redundant if `from_*` methods are used but its to make sure there can't be an `KeyDataSecrets`
+        # instance with an attribute missmatch for calculated cached values. Should be ok since we don't handle a lot of
+        # keys here.
+        try:
+            bytes_from_mnemonic(self.mnemonic_str())
```

Comment:
nit: assign to a local variable since this function is called 3 times?

Reply:
See [adec931](https://github.com/Chia-Network/chia-blockchain/pull/12752/commits/adec93142ac3c4907250f6773d32996766d918cd)